analyzer/codechecker_analyzer/analyzer_context.py

```diff
@@ -53,6 +53,101 @@ def __len__(self):
         return len(self.store)
 
 
+class ProfileMap(Mapping):
+    """
+    A dictionary which maps checker names and checker groups to profile names.
+    A checker or checker group may be the member of multiple profiles.
+    """
+    def __init__(self, profile_map_file):
+        self.store = load_json_or_empty(profile_map_file, {})
+        self.__check_json_format(profile_map_file)
+
+    def __check_json_format(self, profile_map_file):
+        """
+        This function checks the format of checker_profile_map.json config
+        file. If this file doesn't meet the requirements, then CodeChecker
+        exits with status 1.
+
+        {
+          "available_profiles": {
+            "profile1": "description1",
+            "profile2": "description2"
+          },
+          "analyzers": {
+            "analyzer1": {
+              "profile1": ['checker1', 'checker2']
+            },
+            "analyzer2": {
+              "profile1": ['checker3'],
+              "profile2": ['checker3', 'checker4']
+            }
+          }
+        }
+        """
+        if 'available_profiles' not in self.store:
+            LOG.error('Format error in %s: "available_profiles" key not '
+                      'found', profile_map_file)
+            sys.exit(1)
+        if 'analyzers' not in self.store:
+            LOG.error('Format error in %s: "analyzers" key not found',
+                      profile_map_file)
+            sys.exit(1)
+
+        for analyzer, profiles in self.store['analyzers'].items():
+            if not isinstance(profiles, dict):
+                LOG.error('Format error in %s: value of %s must be a '
+                          'dictionary', profile_map_file, analyzer)
+                sys.exit(1)
+
+            diff = set(profiles) - set(self.store['available_profiles'])
+            if diff:
+                LOG.error('Format error in %s: %s at %s not documented under '
+                          '"available_profiles"',
+                          profile_map_file, ', '.join(diff), analyzer)
+                sys.exit(1)
+
+            for profile, checkers in profiles.items():
+                if not isinstance(checkers, list):
+                    LOG.error('Format error in %s: value of %s at analyzer %s '
+                              'must be a list',
+                              profile_map_file, profile, analyzer)
+                    sys.exit(1)
+
+    def __getitem__(self, key):
+        result = []
+        for _, profiles in self.store['analyzers'].items():
```

Comment:
If we don't need the keys of a dict we can use `.values()` to iterate over the values.
```suggestion
        for profiles in self.store['analyzers'].values():
```